creepy/subprocess/pypen.py

```diff
@@ -108,6 +131,26 @@ def detach(self):
             del self._send, self._recv
         except AttributeError:
             pass
+        try:
+            os.remove(self._out_path)
+            os.remove(self._in_path)
+        except (OSError, FileNotFoundError, AttributeError):
+            pass
+
+
+def _make_fifo(path: str | None = None):
+    """Open FIFO at specified path or create and open a new one."""
+    if path is None:
+        path = os.path.join('/tmp', secrets.token_urlsafe(16))
+        os.mkfifo(path)
+    # Open the FIFO for reading. `os.O_NONBLOCK` is used in order to prevent blocking the process when opening.
+    in_fd = os.open(path, os.O_RDONLY | os.O_NONBLOCK)
+    # Since we already opened the FIFO for reading, following open won't block.
+    out_fd = os.open(path, os.O_WRONLY)
+    # Set the file descriptor to just `O_RDONLY` in order to get rid of `O_NONBLOCK` and synchronize the execution of
+    # the two processes.
+    fcntl.fcntl(in_fd, fcntl.F_SETFL, os.O_RDONLY)
```

Comment:
Seems like the task to read and write to the same file should be know, is it standard solution?

Reply:
I haven't found anything. There are plenty examples of how to open pipes using `with open` syntax. However, it uses buffering and is not appropriate in our case.